### ISSUE TITLE
downloader breaking on unicode issues

### ISSUE THREAD

**Issue by alexrudnick** [nltk/nltk#342]
At the tip of master right now, on Python 2.7.3, trying to run nltk.download() and then doing 'u' for update, we get a break due to a unicode issue...

```
>>> import nltk
>>> nltk.download()
NLTK Downloader
---------------------------------------------------------------------------
    d) Download   l) List    u) Update   c) Config   h) Help   q) Quit
---------------------------------------------------------------------------
Downloader> u

Will update following packages (o=ok; x=cancel)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/alex/nltk/nltk/downloader.py", line 645, in download
    self._interactive_download()
  File "/home/alex/nltk/nltk/downloader.py", line 961, in _interactive_download
    DownloaderShell(self).run()
  File "/home/alex/nltk/nltk/downloader.py", line 1000, in run
    self._simple_interactive_update()
  File "/home/alex/nltk/nltk/downloader.py", line 1047, in _simple_interactive_update
    print('  [ ] %s %s' % (pid.ljust(20, '.'), name))
TypeError: must be char, not unicode
```

It also breaks, in a different way, on Python 3.2.3:
    TypeError: unorderable types: Package() < Package()

Working on a fix for both!

**Comment by alexrudnick** [comment]
The surface thing that's going wrong in Python 2.7 is that ljust doesn't work when called on a str with a unicode argument. Like so:

```
>>> "foo".ljust(20, u".")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: must be char, not unicode
```

... and all of our string literals are now unicode. The deeper thing going on is that the strings coming from the server in the index XML are str, not unicode ...

**Comment by kmike** [comment]
Under Python 2.x xml_tree.attrib values are bytestrings for ascii data and unicode for non-ascii data (I think this is a some kind of optimization) so it seems that wrapping values of xml.attrib to compat.text_type (in Package.fromxml) should do the trick.

By the way, I also missed @compat.python_2_unicode_compatible to nltk.downloader.Package (for fixing its `__repr__`).

**Comment by alexrudnick** [comment]
OK! Closed by pull request #344. Thanks @kmike !
